src/megatron/bridge/models/conversion/param_mapping.py

```diff
@@ -795,7 +795,13 @@ def hf_to_megatron(
         megatron_module: nn.Module,
     ) -> torch.Tensor:
         """Replicate weight to all TP ranks."""
-        target_device = megatron_module.weight.device
+        try:
+            target_device = megatron_module.weight.device
+        except AttributeError:
+            # the parameter may not be called "weight"
+            parameter_name = [k for k, v in megatron_module.named_parameters()][0]
+            target_device = getattr(megatron_module, parameter_name).device
```

Comment:
`target_device = next(megatron_module.parameters()).device` to avoid the full list creation?